### PR TITLE
perf: stop counting rows after 10 millions

### PR DESCRIPTION
**Pull request by bajtos**

### CODE REVIEW

File: voyager-publish/test/test.js [hunk 1]
```diff
@@ -24,9 +24,6 @@ describe('unit', () => {
       release () {},
       async query (statement, params) {
         clientStatements.push(statement)
-        if (statement.includes('SELECT COUNT(*) FROM measurements')) {
-          return { rows: [{ count: 10 }] }
-        }
```
Comment (bajtos) [inline voyager-publish/test/test.js:29]: My change did not break the tests, making me believe this block is unnecessary (not used by the tests).

File: voyager-publish/index.js [hunk 2]
```diff
@@ -29,8 +29,10 @@ export const publish = async ({
   // Fetch the count of all unpublished measurements - we need this for monitoring
   // Note: this number will be higher than `measurements.length` because voyager-api adds more
   // measurements in between the previous and the next query.
+  // Note: counting ALL rows can put too much load on the DB server when the table is very large.
+  // Let's stop after we count 10 million rows. That's enough to let us know that we are in a serious trouble.
   const totalCount = (await pgPool.query(
-    'SELECT COUNT(*) FROM measurements'
+    'SELECT COUNT(*) FROM (SELECT 1 FROM measurements LIMIT 10000000) t;'
```
Comment (juliangruber) [inline voyager-publish/index.js:35]:
```suggestion
    'SELECT COUNT(*) FROM (SELECT 1 FROM measurements LIMIT 10000000);'
```

I think that should fix the test

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: voyager-publish/test/test.js (hunk 1 above)
Comment (bajtos) [reply voyager-publish/test/test.js:29]: Weird. The tests are failing in CI.

```  1) unit
       publishes:
     TypeError: Cannot read properties of undefined (reading 'count')
      at publish (file:///home/runner/work/voyager-api/voyager-api/voyager-publish/index.js:36:14)
      at async Context.<anonymous> (file:///home/runner/work/voyager-api/voyager-api/voyager-publish/test/test.js:73:5)
```

File: voyager-publish/index.js [hunk 3]
```diff
@@ -29,8 +29,10 @@ export const publish = async ({
   // Fetch the count of all unpublished measurements - we need this for monitoring
   // Note: this number will be higher than `measurements.length` because voyager-api adds more
   // measurements in between the previous and the next query.
+  // Note: counting ALL rows can put too much load on the DB server when the table is very large.
+  // Let's stop after we count 10 million rows. That's enough to let us know that we are in a trouble.
```
Comment (juliangruber) [inline voyager-publish/index.js:33]:
```suggestion
  // Let's stop after we count 10 million rows. That's enough to let us know that we are in trouble.
```

mince Part II :)

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: voyager-publish/index.js (hunk 2 above)
Comment (bajtos) [reply voyager-publish/index.js:35]:
```
postgres=# SELECT COUNT(*) FROM (SELECT 1 FROM measurements LIMIT 10000000);
ERROR:  subquery in FROM must have an alias
LINE 1: SELECT COUNT(*) FROM (SELECT 1 FROM measurements LIMIT 10000...
                             ^
HINT:  For example, FROM (SELECT ...) [AS] foo.
```

File: voyager-publish/test/test.js (hunk 1 above)
Comment (juliangruber) [reply voyager-publish/test/test.js:29]: see https://github.com/filecoin-station/voyager-api/pull/28#discussion_r1590887825

File: voyager-publish/index.js (hunk 2 above)
Comment (juliangruber) [reply voyager-publish/index.js:35]: ah, then maybe the js side needs to be adjusted

File: voyager-publish/test/test.js (hunk 1 above)
Comment (bajtos) [reply voyager-publish/test/test.js]: Fixed in [6458066](https://github.com/filecoin-station/voyager-api/pull/28/commits/6458066b0d0862a395a590064c3899d998cadfd9) (hopefully)

**Review by juliangruber — APPROVED** [review]

**Pull request merged**